include/openmc/dagmc.h

```diff
@@ -40,7 +40,7 @@ class DAGSurface : public Surface {
   double evaluate(Position r) const override;
   double distance(Position r, Direction u, bool coincident) const override;
   Direction normal(Position r) const override;
-  Direction reflect(Position r, Direction u, Particle* p) const override;
+  Direction reflect(Position r, Direction u) const override;
```

Comment:
Just noticed that the corresponding method signature in `dagmc.cpp` also needs to be updated.

Reply:
ah, thanks! Unfortunately it looks like DAGSurface is the one thing that actually needs a particle pointer to reflect off a surface. Why does DAGMC need particle state to reflect off a surface?

Reply:
Well, not that it matters... I just replaced the Particle* argument with a Geometron* that defaults to nullptr if called without it. Now it should work fine!

Reply:
Coming back to this... we probably still need this argument here. In DAGMC, we can greatly accelerate particle reflection by using the `RayHistory` stored on the `Particle` (now `Geometron`) by using the last triangle hit to perform the reflection. 

The alternative is to search for the nearest triangle to the current location on the reflecting surface, which is what DAGMC will default to a search for the nearest triangle on the surface to the particle's current position, but this operation is slow (search) and less robust if the particle lies on the edge of two triangles with significantly different normals.

I notionally agree that it would be nice to only pass the direction and position to geometric methods like this, but its really useful to have a logical assist because floating point math. FWIW, we pass additional information to the `Cell:distance` method for a similar assist to deal with surface coincidence. That said, I don't love having to pass an entire object to convey this information to the method either b/c 1) It's not as clear what the purpose of passing that object is (i.e. what information its providing to the method) as passing a single value is 2) it's potentially computationally heavy to do this.